### ISSUE TITLE
`pk put` is extremely slow

### ISSUE THREAD

**Issue by surma** [perkeep/perkeep#1625]
I just tried to `put` a folder with ~4000 files, totalling ~640MB. It took 45 minutes(!). I’m looking for help to figure out where I am going wrong.

I tested my network connection, over which `scp` levels out at about 11MB/s.
Disk write speed at `/perkeep-data` is ~80MB/s (it’s a software RAID).
So even with super-conservative back-of-napkin math, I’d expect this to take maybe 10 minutes. Not 45.

My server config is fairly minimal:

```json
{
        "auth": "none",
        "listen": ":3179",
        "camliNetIP": "",
        "identity": "<redacted>",
        "identitySecretRing": "/perkeep/identity-secring.gpg",
        "blobPath": "/perkeep-data/blobs",
        "packRelated": true,
        "levelDB": "/perkeep-data/index.leveldb"
}
```

Anyone know how I can figure out why things are so slow? Do I need to switch away from LevelDB?

**Comment by tgulacsi** [comment]
I'd try packRelated:false, packBlobs:true
That will use diskpacked backend, with fewer bigger files, faster write, little bit slower read rate.

If that's not enough, you could try "eatmydata" (https://www.flamingspork.com/projects/libeatmydata/) to temporarily disable fsync,
to test whether it's the culprit or not.

**Comment by zenhack** [comment]
Re: the latter, I think an LD_PRELOAD binary won't work on Go, since things are statically linked?

But yes, I hit a similar problem when doing an initial import of existing data; diskpacked helped, not fsyncing helped more (I think I wrote a one-off program to run perkeep with a seccomp filter which silently no-oped fsync), and it still took way too long (but I was importing like a TB of data).

**Comment by tgulacsi** [comment]
I've had a PR for fsyncing only each second in diskpacked, but haven't been merged.
For temporary solution, you can try 35765d3b400fb1a3ed5fa84eabe99b3af053dbae - I haven't tested it thoroughly yet.

**Comment by surma** [comment]
I added `packRelated` and `packBlobs` to my config, but even after initiating a full re-scan, all my files have disappeared (scary!). If I remove the config again, my files re-appear. I haven’t dared testing to upload something out of fear of overwriting _something_.

Can I/Do I need to manually unpack the packed blobs to switch this config?

**Comment by tgulacsi** [comment]
I'm afraid yes.
Either you start up another perkeep with different config and `pk sync` between them, or create the new blobstore inside your server (edit the result of `pk dumpconfig` (clone the "/bs/" part as "/bs-dp/" and modify that) and use it as a server-config.json).

**Comment by octaviordz** [comment]
With the configuration of: 
packRelated:false
packBlobs:true

`
storing blobs packed together into monolithic data files`

Does this mean that Perkeep is going to preserve the "file" as a whole?

**Comment by tgulacsi** [comment]
No, it means it uses blobpacked blob store, storing blobs in 512MiB files, packed in arrival order. Each file is append-only, so backup friendly, and uses few vig files instead of many small.
